docker/web-and-data/docker-entrypoint.sh

```diff
@@ -23,19 +23,53 @@ parse_db_params_from_config_and_command_line() {
     else
         PROPERTIES_FILE=$BAKED_IN_WAR_CONFIG_FILE
     fi
-
-    for param in db.host db.user db.portal_db_name db.password; do
+    for param in db.host db.user db.portal_db_name db.password db.connection_string; do
         if $(parse_db_params_from_command_line $@ | grep -q $param); then
-            parse_db_params_from_command_line $@ | grep "^$param"
+            prop=$(parse_db_params_from_command_line $@ | grep "^$param" || [[ $? == 1 ]])
         else
-            grep -v '^#' $PROPERTIES_FILE | grep "^$param"
+            prop=$(grep -v '^#' $PROPERTIES_FILE | grep "^$param" || [[ $? == 1 ]])
+        fi
+        if [[ -n "$prop" ]]
+        then
+            # Replace dot in parameter name with underscore.
+            prop=$(sed "s/^db\./db_/" <<< $prop)
+            if [[ $param == db.connection_string ]]
+            then
+                # Remove the parameters (?...) from the connection URL.
+                echo $(sed -r "s/^([^=]+)=([^\?]+).*/\1=\2/" <<< $prop)
+            else 
+                echo $prop
+            fi
         fi
     done
 }
 
+parse_connection_string() {
+    # Adapted from: https://stackoverflow.com/a/45977232
+    readonly URI_REGEX='^(([^:/?#]+):)+?(//((([^:/?#]+)@)?([^:/?#]+)(:([0-9]+))?))?(/([^?#]*))(\?([^#]*))?(#(.*))?'
+    #                    ↑↑             ↑  ↑↑↑            ↑         ↑ ↑            ↑ ↑        ↑  ↑        ↑ ↑
+    #                    |2 scheme      |  ||6 userinfo   7 host    | 9 port       | 11 rpath |  13 query | 15 fragment
+    #                    1 scheme:      |  |5 userinfo@             8 :…           10 path    12 ?…       14 #…
+    #                                   |  4 authority
+    #                                   3 //…
+    echo db_host=$([[ "$1" =~ $URI_REGEX ]] && echo "${BASH_REMATCH[7]}")
+    echo db_port=$([[ "$1" =~ $URI_REGEX ]] && echo "${BASH_REMATCH[9]}")
+}
+
 check_db_connection() {
-    eval "$(parse_db_params_from_config_and_command_line $@ | sed 's/^db\./db_/g')"
-    POTENTIAL_DB_PARAMS=$@
+    eval $(parse_db_params_from_config_and_command_line $@)
+    
+    if ([[ -n $db_host ]] || [[ -n $db_portal_db_name ]] || [[ -n $db_use_ssl ]]) && [[ -n $db_connection_string ]]
+    then
+        echo "The portal.properties file defines both db.connection_string and (one of) db.host, db.portal_db_name and "
+        echo "db.use_ssl. Please configure with either db.connection_string (preferred), or db.host, db.portal_db_name and db.use_ssl."
+        exit 1
```

Comment:
We're promising backwards-compatibility but that's not true: if staying with properties as they are right now, the error message in line 64 will show up (since we have both obsolete and `db.connection_string` properties defined).

Removing the obsolete properties and leaving `db.connection_string` will not work, since the value of this property needs to be changed too (from `jdbc:mysql://localhost:3306/` to `jdbc:mysql://localhost:3306/cbioportal?useSSL=false`). 

If the user removes the `db.connection_string` and keeps the obsolete properties as they are in their setup, then the _promised_ backwards compatibility comes into play. But since we want to prioritise using the new setup (`db.connection_string`), and the user needs to modify `portal.properties` anyway, maybe it makes more sense to force to use `db.connection_string` with the correct format and clean up the usage of the obsolete properties?

@inodb What do you think?

Reply:
How about we don't allow `db.connection_string` shows up with one of 'db.host', 'db.portal_db_name', and 'db.use_ssl' properties? If that's the case, we throw an error and point out that the users should use `db.connection_string` only instead.

Reply:
I know this is probably not 100% backward compatible, but I guess it's fine.